src/tutorials/data/queries.md

```diff
@@ -0,0 +1,365 @@
+
+
+## Mango queries
+
+The manqo query system was introduced in [CouchDB 2.0](https://blog.couchdb.org/2016/08/03/feature-mango-query/) and offers a declarative JSON syntax to perform queries on documents, inspired from MongoDB.
+
+There are several important concepts to grasp in order to efficiently use mango queries. Here, we give an overview of those concepts and detail common mistakes to avoid. 
+
+
+### Mango queries with cozy-client
+
+[cozy-client](https://github.com/cozy/cozy-client) is a JavaScript library developed by Cozy Cloud that helps to query data to the Cozy back-end, [cozy-stack](https://docs.cozy.io/en/cozy-stack/), without having to manually deal with authentication, index creation, pagination, etc. In the following, we provide cozy-client examples each time we introduce a new concept. If you are not familiar with it, this [tutorial](https://docs.cozy.io/en/cozy-client/getting-started/) is a good starting place. 
+However, if for some reasons you do not want to use cozy-client, it is possible to make queries on your own by directly requesting the [cozy-stack data API](https://docs.cozy.io/en/cozy-stack/data-system/).
+
+In cozy-client, each query is defined as a [Query Definition](https://docs.cozy.io/en/cozy-client/api/cozy-client/#QueryDefinition). This API is helpful to easily chain Mango concepts (selectors, sorts, indexing, …), as we will see in the following sections.
+
+In the following, we assume a DocType named `io.cozy.todos`  describing Todo lists.
+
+Here is a basic example of a cozy-client query returning all the Todo lists in a Cozy:
+```javascript
+const queryDef = client.find("io.cozy.todos")
+const docs = await client.queryAll(queryDef)
+```
+
+In this example, the `QueryDefinition` defines the query to get all the documents stored in the Todos database, expressed through `find("io.cozy.todos")`.
+
+
+## Selectors
+
+To use the Mango query system, you define a `selector`, that allows to express criterias to filter the documents to return.
+
+### Selectors with cozy-client
+
+With cozy-client, a selector is expressed through the [`where`](https://docs.cozy.io/en/cozy-client/api/cozy-client/#querydefinitionwhereselector-querydefinition) method, that takes into input an object with the selector expression. For instance, this query means “the todos in 'sport’ category AND a title ‘Exercices’”: 
+
+```javascript
+const queryDef = client
+  .find("io.cozy.todos")
+  .where({
+    "category": "sport",
+    "title": "Exercices"
+  })
+```
+
+`find` targets the Todos database, while `where` filters the documents to return.
+
+ℹ️ When running this query, an index will automatically be created on `category` and `title` fields. See the [index section](#index-fields) for more details about it. As mentioned in the [performances](#mango-performances), it is highly recommended to read it in order to avoid designing poorly efficient queries.
```

Comment:
```suggestion
ℹ️ When running this query, an index will automatically be created on the `category` and `title` fields. It is highly recommended to read the [index section](#index-fields) to avoid designing poorly efficient queries (see also [the section on mango performances](#mango-performances)). 
```